### DISCUSSION TITLE
Converting SVG to PNG - Depth settings

### DISCUSSION THREAD

**Discussion by kerha66** [dlemstra/Magick.NET#937] [Help]
Hello,

I am trying to convert a SVG file to PNG using the following code:

```
public bool SvgToPngConvertViaMagickNet()
		{
			int dpi = SettingsSvg.Instance.DPI;
			int depth = SettingsSvg.Instance.Depth;
			string pngRelPath = Path.Combine(SvgInfo.RelativeFolderPath, PngRelativPath, Path.GetFileNameWithoutExtension(SvgInfo.ImpFilePath) + ".png");

			MagickReadSettings readSettings = new MagickReadSettings();
			readSettings.Format = MagickFormat.Svg;

			readSettings.Density = new Density(dpi, dpi); // make this configurable
			readSettings.Depth = depth;
			readSettings.Width = (int?) CcscSvg.SvgWidth;
			readSettings.Height = (int?) CcscSvg.SvgHeight;

			try
			{
				Stream svgStream = new MemoryStream();
				svgStream = CcscSvg.CcSvgWriteToStream();

				using (svgStream)
				{
					using (MagickImage image = new MagickImage(svgStream, readSettings))
					{
						try
						 {
							image.Format = MagickFormat.Png;
							image.Density = new Density(dpi,dpi, DensityUnit.PixelsPerInch);
							image.Depth = depth;
							//image.BitDepth(depth);
							//image.Resize(4212,2979); // 4212x2979		2979x2104		2104x1488	1488x1052	1052x744
							//image.Depth = 2;
							//image.Quantize(new QuantizeSettings() { Colors = (int)Math.Pow(2, depth)});

							image.Write(PngPath);
							
							CcscFunctions.LogLine(
								MessageType.Info,
								pngRelPath + " " + ModulSettings.Translation.GetMessageText("image_created"));
						}
						catch (Exception ex)
						{
							if (ex is MagickCorruptImageErrorException || ex is MagickImageErrorException ||
							    ex is MagickDelegateErrorException || ex is MagickMissingDelegateErrorException ||
							    ex is MagickFileOpenErrorException)
							{
								// Not an image, or improper image
								Svgsvg.ErrorQaMessage = Functions.GetFullErrorMessage(ex);
								CcscFunctions.LogLine(
									MessageType.Warning,
									pngRelPath +
									Environment.NewLine +
									Svgsvg.ErrorQaMessage);
								return false;
							}
						}
					}
				}
			}
			catch (Exception ex)
			{
				Svgsvg.ErrorQaMessage = Functions.GetFullErrorMessage(ex);
				CcscFunctions.LogLine(
					MessageType.Warning,
					Svgsvg.ErrorQaMessage);
				return false;
			}

			return true;
		}
```

In the converted PNG the Density (and Colors) is set correctly every time.
I pass different values of Depth but they don't seem to be applied.

I also tried using different values of depth with the Quantize method. Values 4 and 8 worked correctly. Values 2 and 16 didn't seem to create the right output.

So my question is: I am doing something wrong here or is the source file maybe the issue here?

[PARTS_LIST_00533_01_ORG.zip](https://github.com/dlemstra/Magick.NET/files/6531811/PARTS_LIST_00533_01_ORG.zip)

**Comment by dlemstra** [comment]
Not all depths are supported in the PNG format. You don't need to set the depth when reading the image. You only need to set it when writing the image.

**Comment by kerha66** [comment]
Thank you very much for your reply.

I have some follow-up questions:

  1.  When converting from SVG to PNG, if the font size of a text is writing in unit “pt” (or other units than pixel), how does Magick.NET convert those? Are the 'pt' units (or other) converted to 'px' and then the PNG created?


Thank you for your Help 😊

Von: Dirk Lemstra ***@***.***>
Gesendet: Samstag, 29. Mai 2021 11:06
An: dlemstra/Magick.NET ***@***.***>
Cc: Katharina Erhardt ***@***.***>; Author ***@***.***>
Betreff: Re: [dlemstra/Magick.NET] Converting SVG to PNG - Depth settings (#937)


Not all depths are supported in the PNG format. You don't need to set the depth when reading the image. You only need to set it when writing the image.

**Comment by dlemstra** [comment]
The ImageMagick library can only work with pixels. ImageMagick uses a third party library called `librsvg` when converting `SVG` files to pixels that are internally used by ImageMagick. And then those pixels are written to the output image.

**Comment by kerha66** [comment]
Hi Dirk,

All right

  1.  I convert this SVG (in the appendix) to PNG with the Magick.dll.

***@***.***D75E1D.34D4C1E0]

The text 'pt' is used as a font-size unit, in the SVG tags ViewBox, Width and Height do not have a unit.

The texts in the generated PNG are too big. (pixel_from_svg_notransparency.svg)
If there was a solution here.

   2. Another question
The file “V20550000.svg” with dpi=600, depth=4 is converted like this: “V20550000.png”

Thanks and regards



Viele Grüße / Best regards
Katharina Erhardt
Software Developer
[A close up of a signDescription automatically generated]
Quanos Service Solutions GmbH
+49 (9621) 17131-56
***@***.******@***.***>
www.quanos-service-solutions.com<http://www.quanos-service-solutions.com/>
Passion for smart information
***@***.***D761FC.563696F0]<https://hubs.la/H0LzMp50>
Sitz der Gesellschaft: Quanos Service Solutions GmbH - Landsberger Str. 57 - D-82266 Inning am Ammersee
Geschäftsführer: Robert Schäfer, Rafi Boudjakdjian - HRB 99960 (Amtsgericht München) - USt.-IdNr. DE 151 201 914
Die Quanos Service Solutions GmbH steht für fundiertes und zertifiziertes Know-how, frisches Denken und verlässliche Partnerschaft für namhafte Unternehmen aus den verschiedensten Branchen
und bietet professionelle Realisierung von Projekten für interaktive Ersatzteilkataloge und Service-Informationssysteme sowie Serviceportale.
Quanos SIS.one, CATALOGcreator® und PARTS-PUBLISHER sind eingetragene Marken der Quanos Service Solutions GmbH.
Diese Information ist ausschließlich für den Adressaten bestimmt und kann vertrauliche oder gesetzlich geschützte Informationen enthalten.
Wenn Sie nicht der bestimmungsgemäße Adressat sind, unterrichten Sie bitte den Absender und vernichten Sie diese E-Mail.
Anderen als dem bestimmungsgemäßen Adressaten ist es untersagt, diese E-Mail zu lesen, zu speichern, weiterzuleiten oder ihren Inhalt auf welche Weise auch immer zu verwenden.
Wir verwenden aktuelle Virenschutzprogramme. Für Schäden, die dem Empfänger durch von uns zugesandte E-Mails entstehen, schließen wir jede Haftung aus.


Von: Dirk Lemstra ***@***.***>
Gesendet: Donnerstag, 10. Juni 2021 17:06
An: dlemstra/Magick.NET ***@***.***>
Cc: Katharina Erhardt ***@***.***>; Author ***@***.***>
Betreff: Re: [dlemstra/Magick.NET] Converting SVG to PNG - Depth settings (#937)


The ImageMagick library can only work with pixels. ImageMagick uses a third party library called librsvg when converting SVG files to pixels that are internally used by ImageMagick. And then those pixels are written to the output image.

**Comment by dlemstra** [comment]
My message was not send by email, it was posted here: https://github.com/dlemstra/Magick.NET/discussions/937

**Comment by kerha66** [comment]
Hi Dirk,

thanks. 😊


Viele Grüße / Best regards
Katharina Erhardt
Software Developer
[A close up of a signDescription automatically generated]
Quanos Service Solutions GmbH
+49 (9621) 17131-56
***@***.***
www.quanos-service-solutions.com<http://www.quanos-service-solutions.com/>
Passion for smart information
***@***.***D76216.888FD890]<https://hubs.la/H0LzMp50>
Sitz der Gesellschaft: Quanos Service Solutions GmbH - Landsberger Str. 57 - D-82266 Inning am Ammersee
Geschäftsführer: Robert Schäfer, Rafi Boudjakdjian - HRB 99960 (Amtsgericht München) - USt.-IdNr. DE 151 201 914
Die Quanos Service Solutions GmbH steht für fundiertes und zertifiziertes Know-how, frisches Denken und verlässliche Partnerschaft für namhafte Unternehmen aus den verschiedensten Branchen
und bietet professionelle Realisierung von Projekten für interaktive Ersatzteilkataloge und Service-Informationssysteme sowie Serviceportale.
Quanos SIS.one, CATALOGcreator® und PARTS-PUBLISHER sind eingetragene Marken der Quanos Service Solutions GmbH.
Diese Information ist ausschließlich für den Adressaten bestimmt und kann vertrauliche oder gesetzlich geschützte Informationen enthalten.
Wenn Sie nicht der bestimmungsgemäße Adressat sind, unterrichten Sie bitte den Absender und vernichten Sie diese E-Mail.
Anderen als dem bestimmungsgemäßen Adressaten ist es untersagt, diese E-Mail zu lesen, zu speichern, weiterzuleiten oder ihren Inhalt auf welche Weise auch immer zu verwenden.
Wir verwenden aktuelle Virenschutzprogramme. Für Schäden, die dem Empfänger durch von uns zugesandte E-Mails entstehen, schließen wir jede Haftung aus.
Von: Dirk Lemstra ***@***.***>
Gesendet: Dienstag, 15. Juni 2021 17:28
An: dlemstra/Magick.NET ***@***.***>
Cc: Katharina Erhardt ***@***.***>; Author ***@***.***>
Betreff: Re: [dlemstra/Magick.NET] Converting SVG to PNG - Depth settings (#937)


My message was not send by email, it was posted here: #937<https://github.com/dlemstra/Magick.NET/discussions/937>
